stdnum/at/stnr.py

```diff
@@ -0,0 +1,240 @@
+        },
+        '53': {
+            'tax_office': 'Gmunden Vöcklabruck',
+        },
+        '54': {
+            'tax_office': 'Grieskirchen Wels',
+        },
+        '51': {
+            'tax_office': 'Kirchdorf Perg Steyr',
+        },
+        '46': {
+            'tax_office': 'Linz',
+        },
+    },
+    'Salzburg': {
+        '93': {
+            'tax_office': 'Salzburg-Land',
+        },
+        '91': {
+            'tax_office': 'Salzburg-Stadt',
+        },
+        '90': {
+            'tax_office': 'St. Johann Tamsweg Zell am See',
+        },
+    },
+    'Steiermark': {
+        '65': {
+            'tax_office': 'Bruck Leoben Mürzzuschlag',
+        },
+        '72': {
+            'tax_office': 'Deutschlandsberg Leibnitz Voitsberg',
+        },
+        '68': {
+            'tax_office': 'Graz-Stadt',
+        },
+        '69': {
+            'tax_office': 'Graz-Umgebung',
+        },
+        '71': {
+            'tax_office': 'Judenburg Liezen',
+        },
+        '67': {
+            'tax_office': 'Oststeiermark',
+        },
+    },
+    'Tirol': {
+        '81': {
+            'tax_office': 'Innsbruck',
+        },
+        '82': {
+            'tax_office': 'Kitzbühel Lienz',
+        },
+        '83': {
+            'tax_office': 'Kufstein Schwaz',
+        },
+        '84': {
+            'tax_office': 'Landeck Reutte',
+        },
+    },
+    'Vorarlberg': {
+        '97': {
+            'tax_office': 'Bregenz',
+        },
+        '98': {
+            'tax_office': 'Feldkirch',
+        },
+    },
+    'Wien': {
+        '08': {
+            'tax_office': 'Wien 12/13/14 Purkersdorf',
+        },
+        '09': {
+            'tax_office': 'Wien 1/23',
+        },
+        '12': {
+            'tax_office': 'Wien 2/20/21/22',
+        },
+        '03': {
+            'tax_office': 'Wien 3/6/7/11/15 Schwechat Gerasdorf',
+        },
+        '04': {
+            'tax_office': 'Wien 4/5/10',
+        },
+        '06': {
+            'tax_office': 'Wien 8/16/17',
+        },
+        '07': {
+            'tax_office': 'Wien 9/18/19 Klosterneuburg',
+        },
+    },
+
+}
+
+
+def get_all_codes_dict():
+    """Return a dict with opening chars and their tax offices"""
+    offices = {}
+    for _, state_dict in tax_offices_codes.items():
+        offices.update(state_dict)
+    return offices
+
+
+def validate_acceptable_opening_chars(number):
+    """Validate whether the opening chars are acceptable in
+    any tax office"""
+    opening_chars = number[:2]
+    if opening_chars not in get_all_codes_dict():
+        raise InvalidFormat
+    return number
+
+
+def validate_opening_chars_to_office(number, office):
+    """Check if the opening chars are acceptable in that office"""
+    opening_chars = number[:2]
+    office_from_chars = get_all_codes_dict()[opening_chars]['tax_office']
+    if not office_from_chars == office:
+        raise InvalidFormat()
```

Comment:
Wondering if we should provide some error messages for invalid formats?
At least this PR has a few "format is wrong" scenarios, should we specify what exactly went wrong?

Reply:
Makes sense.